unit-tests/func/presets/presets-common.h

```diff
@@ -0,0 +1,332 @@
+
+            auto current = *reinterpret_cast< int32_t * >( (void *)res.data() );
+
+            hw_monitor_command set_command = { librealsense::ivcam2::fw_cmd::AMCSET,
+                                 librealsense::l500_control( op.second ),
+                                 -1 };
+
+            send_command_and_check( dp, set_command );
+
+            // if the sensor is streaming the value of control will update only whan the next
+            // frame arrive
+            if( streaming )
+                std::this_thread::sleep_for( std::chrono::seconds( 50 ) );
+
+            res = send_command_and_check( dp, get_command, 1 );
+            auto def = *reinterpret_cast< int32_t * >( (void *)res.data() );
+            
+            set_command = { librealsense::ivcam2::fw_cmd::AMCSET,
+                            librealsense::l500_control( op.second ),
+                            current };
+
+            send_command_and_check( dp, set_command );
+
+            option_to_defaults[op.first] = float( def );
+        }
+    }
+
+    return option_to_defaults;
+}
+
+std::map< rs2_option, float > get_actual_default_values( rs2::depth_sensor & depth_sens )
+{
+    std::map< rs2_option, float > option_to_defaults;
+
+    for( auto op : option_to_code )
+    {
+        REQUIRE( depth_sens.supports( op.first ) );
+        option_range range;
+        REQUIRE_NOTHROW( range = depth_sens.get_option_range( op.first ) );
+        option_to_defaults[op.first] = range.def;
+    }
+    return option_to_defaults;
+}
+
+void compare( const std::map< rs2_option, float > & first,
+              const std::map< rs2_option, float > & second )
+{
+    CAPTURE( first.size() );
+    CAPTURE( second.size() );
+    REQUIRE( first.size() == second.size() );
+
+    // an order map - should be sorted
+    for( auto it_first = first.begin(), it_second = second.begin(); it_first != first.end();
+         it_first++, it_second++ )
+    {
+        REQUIRE( it_first->first == it_second->first );
+        REQUIRE( it_first->second == it_second->second );
+    }
+}
+
+void print_presets_to_csv( rs2::depth_sensor & depth_sens,
+                           preset_to_expected_values_map & preset_to_expected_values )
+{
+    REQUIRE_NOTHROW( depth_sens.supports( RS2_OPTION_VISUAL_PRESET ) );
+
+    std::ofstream csv( "presets.csv" );
+
+    for_each_preset_mode_combination( [&]( rs2_l500_visual_preset preset, rs2_sensor_mode mode ) {
+        auto expected_values
+            = preset_to_expected_values[{ rs2_l500_visual_preset( preset ), mode }];
+
+        csv << "-----" << rs2_l500_visual_preset( preset ) << " " << rs2_sensor_mode( mode )
+            << "-----"
+            << "\n";
+
+        for( auto i : expected_values )
+        {
+            csv << rs2_option( i.first ) << "," << i.second << "\n";
+        }
+    } );
+
+    csv.close();
+}
+
+void set_values_manually( rs2::depth_sensor & depth_sens, std::map< rs2_option, float > values )
```

Comment:
`const &`
Rename `set_option_values`
Not clear why we pass a `depth_sensor` and not a regular sensor.

Reply:
`values` -> `option_values`